daphne/src/error/mod.rs

```diff
@@ -57,15 +58,27 @@ impl FatalDapError {
 impl From<VdafError> for DapError {
     fn from(e: VdafError) -> Self {
         match e {
-            VdafError::Codec(..) | VdafError::Vdaf(..) => {
+            VdafError::Codec(..) | VdafError::Vdaf(..) | VdafError::Uncategorized(..) => {
                 Self::Transition(TransitionFailure::VdafPrepError)
             }
         }
     }
 }
 
+impl From<CodecError> for DapError {
+    fn from(e: CodecError) -> Self {
+        // Assume this an encoding error and treat as fatal.
+        match e {
+            CodecError::Other(..) => {
+                DapError::Fatal(FatalDapError(format!("encoding failed: {e}")))
+            }
+            unhandled => unreachable!("unhandled error: {unhandled}"),
+        }
+    }
+}
+
```

Comment:
if you're assuming something here than this `From` implementation is only correct for as long as that comment there stays true. Meaning it's holding an invariant over the entire codebase. I'd rather you make something like
```rs
impl DapError {
    fn encoding(e: CodecError) -> Self { ... }
}
```

Reply:
Also, why is only the `Other` variant handled?

Reply:
**Much better**, done!